mysql-test/suite/galera/t/galera_garbd_backup.test

```diff
@@ -0,0 +1,65 @@
+CALL mtr.add_suppression("Process completed with error:.*");
+CALL mtr.add_suppression("Command did not run:.*");
+CALL mtr.add_suppression("WSREP-SST.*");
+
+--echo Starting garbd ...
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9999/xtrabackup_sst//1" --recv-script="sst_garbd_test 9999" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd.log 2>&1 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+--sleep 2 
+
+# Error: incorrect donor sst script
+--error 1
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v3:127.0.0.1:9998/xtrabackup_sst//1" --recv-script="sst_garbd_test 9998" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd2.log 2>&1 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+--sleep 2 
+
+# Error: incorrect receiver script
+--error 2
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9999/xtrabackup_sst//1" --recv-script="not_sst_garbd_test 9999" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd2.log 2>&1 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+#--echo Restarting node #2 to satisfy MTR's end-of-test checks
+#--connection node_2
+#--source include/start_mysqld.inc
+
+# Simulate two connection error 
+
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9999/xtrabackup_sst//1" --donor node1 --recv-script="sst_garbd_test 9999" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd3.log 2>&1 & 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+# Depending on how much time it passes since the previous garbd start, this either fails or succeeds.
+# More than a few ms: the second garbd process knows abaut the SST transfer for the first, and waits until it is done
+# Minimal time: both garbd processes think that the donor sends data to them.
+# Second garbd process only notices the issue when the SST process finishes yet it didn't receive any data, and it exists with a failure.
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9998/xtrabackup_sst//1" --donor node1 --recv-script="sst_garbd_test 9998" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd4.log 2>&1  || exit 0
+
+--connection node_1
+
+# Wait until the background garbd process completes to avoid shutdown assertion
```

Comment:
Can we use start_proc_in_background.inc to start garbd and then wait_proc_to_finish.inc?

Reply:
Good idea, done.